kubert/src/admin.rs

```diff
@@ -238,3 +250,22 @@ fn handle_ready(Readiness(ready): &Readiness, req: Request<Body>) -> Response<Bo
             .unwrap(),
     }
 }
+
+fn handle_metrics(
+    metrics: &PrometheusHandle,
+    process: &Collector,
+    req: Request<Body>,
+) -> Response<Body> {
+    process.collect();
+    match *req.method() {
+        hyper::Method::GET | hyper::Method::HEAD => Response::builder()
+            .status(hyper::StatusCode::OK)
+            .header(hyper::header::CONTENT_TYPE, "text/plain")
```

Comment:
it would be kind of nice to add GZIP compression support here, which we can probably more or less copy directly from the implementation in the `linkerd2-proxy` repo: https://github.com/linkerd/linkerd2-proxy/blob/727f23e20d54003e729ade38c2cefb0762d42f81/linkerd/metrics/src/serve.rs#L21-L55

but, that would probably be best left to a follow-up PR.